TCC.Lib/Benchmark/BenchmarkOptionHelper.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using TCC.Lib.Command;
+using TCC.Lib.Options;
+
+namespace TCC.Lib.Benchmark
+{
+    public static class BenchmarkOptionHelper
+    {
+
+        public static PasswordOption GenerateDecompressPasswordOption(PasswordMode passwordMode, string keysFolder)
+        {
+            switch (passwordMode)
+            {
+                case PasswordMode.None:
+                    return NoPasswordOption.Nop;
+                case PasswordMode.InlinePassword:
+                    return new InlinePasswordOption { Password = "1234" };
+                case PasswordMode.PasswordFile:
+                    string passfile = Path.Combine(keysFolder, "password.txt");
+                    return new PasswordFileOption { PasswordFile = passfile };
+                case PasswordMode.PublicKey:
+                    return new PrivateKeyPasswordOption
+                    {
+                        PrivateKeyFile = Path.Combine(keysFolder, "private.pem")
+                    };
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(passwordMode), passwordMode, null);
+            }
+        }
+
+        public static async Task<PasswordOption> GenerateCompressPassswordOption(PasswordMode passwordMode, string keysFolder)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this method into two, one handling parameters check and the other handling the iterator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS4457)